### ISSUE TITLE
Can't import std lib module for example Deno script

### ISSUE THREAD

**Issue by jimjam-slam** [quarto-dev/quarto-cli#3723]
### Bug description

I'm trying to run the [example Deno script code](https://quarto.org/docs/projects/scripts.html#deno-scripts) inside a Quarto project (that does have a `_quarto.yml`).

Here's my script, `snel.ts`:

```ts
import { parse } from "https://deno.land/std/encoding/yaml.ts";

const config = parse(Deno.readTextFileSync("_quarto.yml"));
```

When I ran this on Quarto 1.3.21, I got an error:

```
$ quarto run snel.ts
error: Specifier not found in cache: "https://deno.land/std@0.153.0/encoding/yaml.ts", --cached-only is specified.
    at file:///[project path]/snel.ts:1:23
```

I understand that Quarto's Deno can't download modules, but I would assume that this module is one of the cached ones.

I've now upgraded to Quarto 1.3.56, but I'm getting a similar-sounding error:

```
$ quarto run snel.ts
error: Module not found "file:///Users/jgol0005/Applications/quarto/vendor/deno.land/std@0.166.0/encoding/yaml.ts".
    at file:///[project path]/snel.ts:1:23
```

Having a look at the contents of the newly installed Quarto, it looks like the path relative to the Quarto installation should probably be `/bin/vendor/deno-land` rather than `/vendor/deno.land`, although I can't see `encoding/yaml.ts` in any of the cached `std@[version]` directories:

![Screen Shot 2022-12-21 at 11 17 12](https://user-images.githubusercontent.com/6520659/208791677-4deb2547-c2c3-4c76-99c5-863d5878e224.png)

Does Quarto manage filling this cache on demand, or should the full standard library be there on install?

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
There's a different place where the stdlib caching for `deno run` is stored. In the source tree, that lives in `package/dist/share/deno_std`, so that's likely in your `share` subdirectory. That gets resolved differently than the libraries you're seeing.

(That doesn't mean this isn't a bug, I'm just giving context.)

**Comment by cscheid** [comment]
I can't currently reproduce this on my Mac either, unfortunately. I added a `console.log(JSON.stringify(config, null, 2))` line and got this:

```
quarto run snel.ts
{
  "project": {
    "type": "website"
  },
  "website": {
    "title": ".",
    "navbar": {
      "left": [
        {
          "href": "index.qmd",
          "text": "Home"
        },
        "about.qmd"
      ]
    }
  },
  "format": {
    "html": {
      "theme": "cosmo",
      "css": "styles.css",
      "toc": true
    }
  }
}
```

**Comment by jimjam-slam** [comment]
Makes sense! I upgraded over the top, so I might do a clean install and see what happens 😊

**Comment by cscheid** [comment]
@jimjam-slam Just checking in on this. Our upgrade should have happened cleanly, but I'm wondering if it's a different bug.

**Comment by jimjam-slam** [comment]
Sorry, @cscheid! I swanned off on holiday and forgot all about it 😅 I'll see what I can do!

**Comment by cscheid** [comment]
@jimjam-slam no hurry! I'm technically out as well :) I just wanted to make I hadn't dropped the ball somewhere either.

**Comment by jimjam-slam** [comment]
I had a crack at "clean installing" by deleting the quarto folder at `~/Applications/quarto` (I'd opted for single-user install), installed Quarto 1.3.78, opened a new shell and tried this again. Still no dice though, unfortunately!

(note: `testscript` is the project folder below)

```shell
$ quarto run snel.ts
error: Module not found "file:///Users/[username]/Applications/quarto/vendor/deno.land/std@0.166.0/encoding/yaml.ts".
    at file:///Users/[username]/Code/svelte/testscript/snel.ts:1:23
$ which quarto
/Users/[username]/Applications/quarto/bin/quarto
```

**Comment by jimjam-slam** [comment]
Oh! I deleted the quarto folder again and reinstalled Quarto 1.3.78 with an all-user install (putting it in `/usr/local/bin/quarto` instead of `~/Applications/quarto` and the command ran successfully, producing the expected JSON output on line 3.

Looks like this might something to do with the single user install!

**Comment by jimjam-slam** [comment]
Apologies—this has cropped up for me again as I've attempted to move up to Quarto 1.3.107.

I've deleted `/Applications/quarto` and `/usr/local/bin/quarto` (which looked like it symlinked to `/Applications/quarto/bin/quarto`—I hadn't clocked that in my previous post), then run the macOS installer, again choosing all-user install.

With a fresh terminal, I can see the symlink again pointing to `/Applications/quarto/bin/quarto` (which I can see was just created), but when I `quarto render` a project tht was previously working, I get:

> error: Module not found "file:///Applications/quarto/vendor/deno.land/std@0.166.0/path/mod.ts".

(The offending line is `import * as path from "https://deno.land/std/path/mod.ts";`

**Comment by jimjam-slam** [comment]
I should note that if I downgrade to 1.2.313, I no longer get this error (even if I install it in the same way, using the `.pkg` installer and specifying "all-user" install).

**Comment by cscheid** [comment]
I'm pretty sure what's happening is that we're letting deno create a cache somewhere, and then we're resolving to the wrong cache. You're seeing that problem because we need to switch deno stdlibs when we switch deno version, and I bet deno doesn't version its internal caches (as I would argue it should)

**Comment by jimjam-slam** [comment]
Let me know if there's anything I can do on my side to help you isolate it!

**Comment by cderv** [comment]
I confirmed that https://github.com/quarto-ext/shinylive/issues/8 is now fixed using very latest 1.3

**Comment by cscheid** [comment]
@jimjam-slam I'm going to go ahead and close this one. If the latest prerelease continues to give you trouble, please feel free to reopen this!
